src/neurostatslib/utils.py

```diff
@@ -4,44 +4,263 @@
+            f"The shape of basis_matrix is {basis_matrix.shape}!"
+        )
+    try:
+        # this should fail for variable trial length
+        time_series = jnp.asarray(time_series)
+        if time_series.ndim != 3:
+            raise ValueError
+
+    except ValueError:
+        # convert each trial to array
+        time_series = [jnp.asarray(trial) for trial in time_series]
+        if not check_dimensionality(time_series, 2):
+            raise ValueError(
+                "time_series must be an iterable of 2 dimensional array-like objects."
+            )
+
+    if any(k == 0 for trial in time_series for k in trial.shape) or (
+        len(time_series) == 0
+    ):
+        raise ValueError("time_series should not contain empty trials!")
+
+    # Check window size
+    ws = len(basis_matrix)
+    if any(trial.shape[0] < ws for trial in time_series):
+        raise ValueError(
+            "Insufficient trial duration. The number of time points in each trial must "
+            "be greater or equal to the window size."
+        )
+
+    # Check if all trials have the same duration # check if needed
+    same_dur = time_series.ndim == 3 if isinstance(time_series, jnp.ndarray) else False
+
+    if same_dur:
+        # Trials have the same duration. # call conv instead of corr
+        conv_trials = list(_CORR_SAME_TRIAL_DUR(time_series, basis_matrix))
+    else:
+        # Trials have variable durations.
+        conv_trials = [
+            _CORR_VARIABLE_TRIAL_DUR(jnp.atleast_2d(trial), basis_matrix)
+            for trial in time_series
+        ]
+
+    return conv_trials
+
+
+def pad_dimension(
+    array: jnp.ndarray,
+    axis: int,
+    window_size: int,
+    filter_type: str = "causal",
+    constant_values: float = jnp.nan,
+) -> jnp.ndarray:
+    """
+    Add padding to the last dimension of an array based on the convolution type.
+
+    Parameters
+    ----------
+    array:
+        The array to be padded.
+    axis:
+        The axis to be padded.
+    window_size:
+        The window size to determine the padding.
+    filter_type:
+        The type of convolution, default is 'causal'. It must be one of 'causal', 'acausal', or 'anti-causal'.
+    constant_values:
+        The constant values for padding, default is jnp.nan.
+
+    Returns
+    -------
+    :
+        An array with padded last dimension.
+    """
+    if axis < 0 or not isinstance(axis, int):
+        raise ValueError("`axis` must be a non negative integer.")
+    elif axis >= array.ndim:
+        raise IndexError(
+            "`axis` must be smaller than `array.ndim`. "
+            f"array.ndim is {array.ndim}, axis = {axis} provided!"
+        )
+
+    padding_settings = {
+        "causal": (window_size, 0),
+        "acausal": ((window_size - 1) // 2, window_size - 1 - (window_size - 1) // 2),
+        "anti-causal": (0, window_size),
+    }
+
+    pad_width = (
+        ((0, 0),) * axis
+        + (padding_settings[filter_type],)
```

Comment:
should raise a more informative error message if `filter_type` isn't one of the expected values.

Reply:
done